src/cachemachine/salcyclerepoman.py

```diff
@@ -0,0 +1,23 @@
+from cachemachine.rubinrepoman import RubinRepoMan
+from cachemachine.rubinimageinfo import RubinImageInfo
+from typing import Dict, Any
+
+
+class SALCycleRepoMan(RubinRepoMan):
+    def __init__(self, body: Dict[str, Any]):
+        """Create a new SALCycleRepoMan.  It's the same as the parent except
+        that the body passed to it may have an additional parameter 'cycle'
+
+        body:
+            cycle: Which XML Cycle to allow.  If specified, filter out
+                   any images that don't have a matching cycle tag.
+        """
+        super().__init__(body)
+        self.cycle = body.get("cycle", None)
+
+    def _allow_image(self, img: RubinImageInfo) -> bool:
+        """If self.cycle was defined at instance creation, only allow images
+        whose cycle matches."""
+        if not self.cycle:
+            return True
+        return img.cycle == self.cycle
```

Comment:
Since this class is so simple, it seems more complex to have it as a subclass rather than just an optional parameter to RubinRepoMan, since we're changing that a lot too

Reply:
If you're happy with that I'm happy with that.  That wasn't what I understood you to be asking for when I brought up the original proposal, is all.